internal/clisnapshot/snapshot.go

```diff
@@ -10,13 +12,11 @@ import (
 	pb "github.com/hashicorp/waypoint/internal/server/gen"
 )
 
-type Config struct {
-	// This Client points to a project's client, as set in the baseCommand
-	Client pb.WaypointClient
-}
-
-func (c *Config) WriteSnapshot(ctx context.Context, w io.Writer) error {
-	stream, err := c.Client.CreateSnapshot(ctx, &emptypb.Empty{})
+// WriteSnapshot accepts a context, WaypointClient, and io.Writer, and returns
+// an error. It uses the Client to create a snapshot and write it to the 
+// provided writer.
```

Comment:
This is _totally fine_ but in the interest of sharing my viewpoints (you don't have to change this at all just sharing my opinion): you don't need to reiterate the args cause we can see that in the signature, so something like below would be enough in this case.

> WriteSnapshot requests a snapshot from the client and writes it to the provider writer.

I usually will try to add details that may be helpful that the signature doesn't show, such as:

> Cancelling the context will prematurely cancel the snapshot. This may result in partial
> writes to the writer.

In that case, knowing this function may partially write when cancelled (and that it can be canceled) _might_ be useful. Again, you don't have to add this but just showing an example of my thought process around docs/comments.

Reply:
All good! That's helpful background on godoc strategy. It did feel repetitive.